stock_quantity_history_location/models/stock_quant.py

```diff
@@ -0,0 +1,28 @@
+# Copyright 2024 ForgeFlow S.L.
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from lxml import etree
+
+from odoo import api, models
+
+
+class StockQuant(models.Model):
+    _inherit = "stock.quant"
+
+    @api.model
+    def get_views(self, views, options=None):
+        res = super().get_views(views, options=options)
+        user_has_group = self.user_has_groups("stock.group_stock_multi_locations")
+
+        if user_has_group:
+            list_view = res.get("views", {}).get("list", {})
+            if list_view:
+                arch = list_view.get("arch", "")
+                arch_tree = etree.XML(arch)
+                buttons = arch_tree.xpath('//button[@name="action_inventory_at_date"]')
+                for button in buttons:
+                    button.set("string", "Inventory at Date & Location")
+                new_arch = etree.tostring(arch_tree, encoding="unicode")
+                list_view["arch"] = new_arch
+
+        return res
```

Comment:
```suggestion
    @api.model
    def get_views(self, views, options=None):
        res = super().get_views(views, options=options)
        user_has_group = self.user_has_groups("stock.group_stock_multi_locations")
        list_view = res.get("views", {}).get("list", {})
        if user_has_group and list_view:
            arch = list_view.get("arch", "")
            arch_tree = etree.XML(arch)
            buttons = arch_tree.xpath('//button[@name="action_inventory_at_date"]')
            for button in buttons:
                button.set("string", "Inventory at Date & Location")
            new_arch = etree.tostring(arch_tree, encoding="unicode")
            list_view["arch"] = new_arch
        return res
```
We can do both checks at the beginning, and then we can remove one indentation, just as a smaller improvement. Well resolved 👍🏿